src/main/java/platform/codingnomads/co/springweb/gettingdatafromclient/handlingmultipartdata/controllers/HandleMultipartDataController.java

```diff
@@ -95,6 +97,28 @@ public ResponseEntity<?> downloadFileById(@PathVariable(name = "id") Long fileId
                 .body(new ByteArrayResource(databaseFile.getData()));
     }
 
+    // GET file by file name
+    @GetMapping("/searchByName/{fileName}")
+    public ResponseEntity<?> searchFilesByName(@PathVariable String fileName) {
+
+        final boolean existsByFileName = fileRepository.existsByFileName(fileName);
+
+        if (!existsByFileName) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("File not found with file name: " + fileName);
+        }
+
+        List<DatabaseFile> databaseFiles = fileRepository.findByFileName(fileName);
+
+        List<FileResponse> fileResponses = new ArrayList<>();  // Is there a more efficient way to do this?
+        for (DatabaseFile databaseFile : databaseFiles) {
+            FileResponse fileResponse = FileResponse.builder().fileName(databaseFile.getFileName())
+                    .fileType(databaseFile.getFileType()).build();
+            fileResponses.add(fileResponse);
+        }
+
+        return ResponseEntity.ok(fileResponses);
+    }
```

Comment:
Since you need to remap one type `DatabaseFile` to another `FileResponse` it should be done manually, so your solution is the way. 
However, there are could be a couple of ways to make it closer to production-ready code:
- Extract the mapping to a separate "MappingUtil" class and use it. It would be a faster way. 
- Use [ModelWrapper](https://modelmapper.org/getting-started/) it will analyze `from` and `to` objects and map fields that have the same type nad name automatically (you can also provide some instructions on how to map fiels with different names) It is powerfull library, here [some more deep dives](https://www.baeldung.com/java-modelmapper) into functionality. It would a bit slower way since it uses a library with a bunch of `reflection` invocations.

A nit, you don't need to create intermediate objects, so the implementation could look like:
```java 
    @GetMapping("/searchByName/{fileName}")
    public ResponseEntity<?> searchFilesByName(@PathVariable String fileName) {
        if (!fileRepository.existsByFileName(fileName)) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("File not found with file name: " + fileName);
        }

        List<FileResponse> fileResponses = new ArrayList<>();  // Is there a more efficient way to do this?
        for (DatabaseFile databaseFile : fileRepository.findByFileName(fileName)) {
            fileResponses.add(FileResponse.builder()
                    .fileName(databaseFile.getFileName())
                    .fileType(databaseFile.getFileType())
                    .build());
        }

        return ResponseEntity.ok(fileResponses);
    }
```

Reply:
ModelWrapper, very cool!